src/DI/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -26,6 +26,11 @@ public ServiceProviderEngineScope(ServiceProviderEngine engine)
 
         public object GetService(Type serviceType)
         {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(IServiceScope));
```

Comment:
Why change patterns from ServiceProviderEngine and throw ODE directly from the method?

Reply:
Good point.